src/GenerateYotiLetterHandler.ts

```diff
@@ -2,35 +2,63 @@ import { Logger } from "@aws-lambda-powertools/logger";
 import { Metrics } from "@aws-lambda-powertools/metrics";
 import { LambdaInterface } from "@aws-lambda-powertools/commons";
 import { Constants } from "./utils/Constants";
+import { AppError } from "./utils/AppError";
+import { HttpCodesEnum } from "./utils/HttpCodesEnum";
+import { MessageCodes } from "./models/enums/MessageCodes";
+import { getParameter } from "./utils/Config";
+import { Response } from "./utils/Response";
+import { GenerateYotiLetterProcessor } from "./services/GenerateYotiLetterProcessor";
+import { EnvironmentVariables } from "./services/EnvironmentVariables";
+import { ServicesEnum } from "./models/enums/ServicesEnum";
 
 const {
 	POWERTOOLS_METRICS_NAMESPACE = Constants.F2F_METRICS_NAMESPACE,
 	POWERTOOLS_LOG_LEVEL = Constants.DEBUG,
-	POWERTOOLS_SERVICE_NAME = "TODO",
+	POWERTOOLS_SERVICE_NAME = Constants.GENERATE_YOTI_LETTER_SVC_NAME,
 } = process.env;
 
 
-const logger = new Logger({
+export const logger = new Logger({
 	logLevel: POWERTOOLS_LOG_LEVEL,
 	serviceName: POWERTOOLS_SERVICE_NAME,
 });
 
-const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
+let yotiPrivateKey: string;
+
+const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE });
+
+export class GenerateYotiLetterHandler implements LambdaInterface {
+	private readonly environmentVariables = new EnvironmentVariables(logger, ServicesEnum.GENERATE_YOTI_LETTER_SERVICE);
 
-class GenerateYotiLetterHandler implements LambdaInterface {
 	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
-	handler(event: any, context: any): any {
+	async handler(event: any, context: any): Promise<any> {
```

Comment:
I think we should validate the event has both the required attributes before we proceed with the processor as both are mandatory and required. Good to validate the event else throw an error.

Reply:
Thanks for taking care of this.